app/services/paypal/refresh_identity.rb

```diff
@@ -0,0 +1,51 @@
+class Paypal::RefreshIdentity < BaseService
+  def initialize(publisher_id:)
+    @publisher = Publisher.find(publisher_id)
+    @paypal_connection = @publisher.paypal_connection
+  end
+
+  def perform
+    return if @paypal_connection.nil? || @paypal_connection.refresh_token.nil?
+    begin
+      access_token = fetch_access_token
+      raise_token_exception if access_token.nil?
+    rescue => e
+      Raven.capture_exception(e)
+    end
+
+    user_info_response = Faraday.get("#{Rails.application.secrets[:paypal_api_uri]}/v1/identity/oauth2/userinfo") do |req|
+      req.headers['Authorization'] = "Bearer #{access_token}"
+      req.headers['Accept'] = 'application/json'
+      req.params['schema'] = "paypalv1.1"
+    end
+
+    user_info = JSON.parse(user_info_response.body)
+    paypal_connection = PaypalConnection.find_by(
+      user_id: @publisher.id,
+      hidden: false
+    )
+    paypal_connection.update(
+      country: user_info.dig('address', 'country'),
+      verified_account: user_info.dig('verified_account'),
+      paypal_account_id: user_info.dig('user_id')
+    )
+  end
+
+  private
+
+  def raise_token_exception
```

Comment:
Nit methods that raise exceptions should have exclamation marks 
```suggestion
  def raise_token_exception!
```